src/main/java/edu/ksu/canvas/requestOptions/SubmissionOptions.java

```diff
@@ -0,0 +1,124 @@
+package edu.ksu.canvas.requestOptions;
+
+import java.util.Map;
+
+/**
+ * Rubric_Assessment not supported yet.
+ */
+public class SubmissionOptions extends BaseOptions {
+
+    public class StudentSubmissionOption {
+        private String textComment;
+        private String postedGrade;
+        private Boolean excuse;
+        private Boolean groupComment;
+        private String mediaCommentId;
+        private String mediaCommentType;
+
+        public StudentSubmissionOption(String textComment, String postedGrade, Boolean excuse, Boolean groupComment, String mediaCommentId, String mediaCommentType) {
+            this.textComment = textComment;
+            this.postedGrade = postedGrade;
+            this.excuse = excuse;
+            this.groupComment = groupComment;
+            this.mediaCommentId = mediaCommentId;
+            this.mediaCommentType = mediaCommentType;
+        }
+
+
+        public String getTextComment() {
+            return textComment;
+        }
+
+        public void setTextComment(String textComment) {
+            this.textComment = textComment;
+        }
+
+        public String getPostedGrade() {
+            return postedGrade;
+        }
+
+        public void setPostedGrade(String postedGrade) {
+            this.postedGrade = postedGrade;
+        }
+
+        public Boolean getExcuse() {
+            return excuse;
+        }
+
+        public void setExcuse(Boolean excuse) {
+            this.excuse = excuse;
+        }
+
+        public Boolean getGroupComment() {
+            return groupComment;
+        }
+
+        public void setGroupComment(Boolean groupComment) {
+            this.groupComment = groupComment;
+        }
+
+        public String getMediaCommentId() {
+            return mediaCommentId;
+        }
+
+        public void setMediaCommentId(String mediaCommentId) {
+            this.mediaCommentId = mediaCommentId;
+        }
+
+        public String getMediaCommentType() {
+            return mediaCommentType;
+        }
+
+        public void setMediaCommentType(String mediaCommentType) {
+            this.mediaCommentType = mediaCommentType;
+        }
+    }
+
+    private Integer courseId;
```

Comment:
Currently this `courseId` variable is never used since you have only implemented the section level call. However in my comment above I suggested implementing the course level call as well. I think you could just take a single Integer here and name it `objectId` and specify in the constructor's javadoc comments (in an @params annotation) that it must either be a course or section ID. I did something similar in the [GetEnrollmentOptions](/kstateome/canvas-api/blob/master/src/main/java/edu/ksu/canvas/requestOptions/GetEnrollmentOptions.java) class which can also be applied to a couple of different Canvas object types.

Also, I am pretty sure the section or course IDs in these API calls could be SIS IDs (by using the string `sis_course_id:1234` instead of the Canvas course ID) so it should probably be a String instead of an Integer.